provider/linode.go

```diff
@@ -90,7 +90,9 @@ func NewLinodeProvider(domainFilter DomainFilter, dryRun bool) (*LinodeProvider,
 		},
 	}
 
+	ua := fmt.Sprintf("ExternalDNS %s (linodego %s)", appVersion, linodego.Version)
```

Comment:
Maybe nitpick, but the User Agent usually looks like "MyApp/1.0" (e.g. "Mozilla/5.0" or https://github.com/hjacobs/kube-resource-report/blob/master/kube_resource_report/report.py#L66).

Reply:
I actually think getting a good format is really important. Cc @displague. We were discussing this earlier

Reply:
Some more inspiration: https://perishablepress.com/list-all-user-agents-top-search-engines/

Reply:
I think `ExternalDNS/%s (linodego %s)` or `ExternalDNS/%s linodego/%s` would both be acceptable, but I would lean towards the second one.

For example,
http://svn.tools.ietf.org/svn/wg/httpbis/specs/rfc7231.html#header.user-agent
```
  User-Agent: CERN-LineMode/2.15 libwww/2.17b3
```

I was somewhat surprised that other API clients don't have their User-Agents set in this project.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent#Firefox_UA_string

The parenthesis field may be intended to describe the operating environment of the client.  Perhaps the Kubernetes version?  I don't think we need/want to send that much information.

Reply:
Changed to ExternalDNS/%s linodego/%s